osrsbox/items_api/all_items.py

```diff
@@ -63,43 +64,46 @@ def __len__(self) -> int:
         """
         return len(self.all_items)
 
-    def load_all_items(self, input_data_file_or_directory: str) -> None:
+    def load_all_items(self, input_data_file_or_directory: Path) -> None:
         """Load the items database via a JSON file, or directory of JSON files.
 
         :param input_data_file_or_directory: The path to the data input.
         """
-        if os.path.isdir(input_data_file_or_directory):
-
-            if input_data_file_or_directory.endswith("/"):
-                path = input_data_file_or_directory + "*"
-            else:
-                path = os.path.join(input_data_file_or_directory, "*")
-
-            self._load_items_from_directory(path_to_directory=path)
-
-        elif os.path.isfile(input_data_file_or_directory):
-            self._load_items_from_file(input_data_file_or_directory)
+        # Check if a str is supplied, if so, convert to Path object
+        if isinstance(input_data_file_or_directory, str):
+            input_data_file_or_directory = Path(input_data_file_or_directory)
+
+        # Process the directory of JSON, or a single JSON file
+        if input_data_file_or_directory.is_dir():
+            self._load_items_from_directory(path_to_directory=input_data_file_or_directory)
+        elif input_data_file_or_directory.is_file():
+            self._load_items_from_file(path_to_json_file=input_data_file_or_directory)
+        else:
+            raise SystemExit("Error: Valid input not found. Exiting.")
```

Comment:
I would not suggest `raise SystemExit` from any code that will be ran by clients that is not on the command line. Someone might want to use this method and handle the error instead of us trying to force the exit. I think you should just  throw this exception as some other type. `ValueError` is for when arguments are incorrectly supplied so I think it would be best to use that.

BTW, adding more exceptions is good, you might want to start using the `:raises Exception:` docstring when you throw errors. For instance, throwing `ValueError`:
```python
def my_method():
    """
    :raises ValueError: if we cannot create a class based on the input provided
    """
try:
    method()
except TypeError as e:
    raise ValueError("Unable to create class with parameters provided") from e
```

Reply:
Thanks, this sounds like good practice. I will make the change as suggested.